lib/jortt/client/customers.rb

```diff
@@ -8,76 +8,84 @@ class Client # :nodoc:
     #
     # @see { Jortt::Client.customers }
     class Customers
+      attr_accessor :token
 
-      def initialize(config)
-        @resource = RestClient::Resource.new(
-          "#{config.base_url}/customers",
-          user: config.app_name,
-          password: config.api_key,
-        )
+      def initialize(token)
+        @token = token
       end
 
       ##
-      # Returns all customers using the GET /customers/all endpoint.
+      # Returns all customers using the GET /customers endpoint.
       #
       # @example
-      #   Jortt::Client.customers.all(page: 3, per_page: 10)
+      #   client.customers.index(page: 3, query: 'Jane')
       #
-      def all(page: 1, per_page: 50)
-        resource['all'].get(params: {page: page, per_page: per_page}) do |res|
-          JSON.parse(res.body)
-        end
+      def index(page: 1, query: nil)
+        token.get('/customers', params: {page: page, query: query}).parsed.fetch('data')
```

Comment:
You could DRY up the `token.get(path, params).parse.fetch('data')` since that seems repeated a lot.